app/api-v1/controllers/Attachment/index.js

```diff
@@ -1,5 +1,73 @@
+const { parseAccept } = require('../../../utils/parseAcceptHeader')
+const logger = require('../../../utils/Logger')
+
+const db = require('../../../db')
+
+const { createAttachmentFromFile, createAttachment } = require('../Attachment/helpers')
+const { InternalError, NotFoundError, BadRequestError, NotAcceptableError } = require('../../../utils/errors')
+
 module.exports = {
   getAll: async function () {
     return { status: 500, response: { message: 'Not Implemented' } }
   },
+  get: async (req) => {
+    const { id } = req.params
+    if (!id) throw new BadRequestError('missing params')
+
+    const [attachment] = await db.getAttachment(req.params.id)
+    if (!attachment) throw new NotFoundError('Attachment Not Found')
+    const orderedAccept = parseAccept(req.headers.accept)
+
+    for (const mimeType of orderedAccept) {
+      if (mimeType === 'application/octet-stream' && attachment.filename.includes('json')) {
+        throw new NotAcceptableError('Client file request not supported')
+      }
+
+      if (mimeType === 'application/json' || mimeType === 'application/*' || mimeType === '*/*') {
+        const json = JSON.parse(attachment.binary_blob)
+        return {
+          status: 200,
+          response: json,
+        }
+      }
+      if (mimeType === 'application/octet-stream') {
+        return {
+          status: 200,
+          headers: {
+            immutable: true,
+            maxAge: 365 * 24 * 60 * 60 * 1000,
+            'content-disposition': `attachment; filename="${attachment.filename}"`,
+            'access-control-expose-headers': 'content-disposition',
+            'content-type': 'application/octet-stream',
+          },
+          response: attachment.binary_blob.toString(),
```

Comment:
Pretty sure this should not be doing a `toString()`. I commented before how that likely results in an invalid utf8 conversaion

Reply:
why has this been resolved? I don't believe this was updated?

Reply:
sorry my changes have gone to wrong branch